src/main/java/com/gard/testbed/activitySystem/State.java

```diff
@@ -16,21 +16,32 @@ public State(String stateId) {
     }
 
     @Override
-    public String getId() {
-        return id;
+    public boolean isCompleted(IBlackboard player) {
+        return transitions.stream().allMatch(e -> e.isComplete(player));
+    }
+
+    @Override
+    public boolean isActive() {
+        return false;
     }
```

Comment:
The assumption that a state is complete when all its incoming transitions are satisfied is pretty strong. You are actually saying that the transitions once true they will stay true in the future. Which can lead to problems with ciclic graphs